docs/design/v2/delete-functionality.md

```diff
@@ -0,0 +1,157 @@
+an ImageSetConfig, accidentally deleting wanted (deployed) images.
+
+A new oc-mirror delete sub command is now available to execute the delete workflow with the DeleteImageSetConfiguration input.
+
+This functionality is able to work in a complete "air-gapped" (disconnected) environment, provided that the images have been
+previously mirrored (i.e the local cache and remote registry know about the images being deleted).
+
+The delete functionality is split into 2 stages:
+
+- The first stage is to create (generate) a delete yaml file using the --generate flag, this file is used to validate the images/blobs that will be deleted.
+- The second stage is to use the delete yaml file (once validated) to actually perform a local cache and remote registry delete.
+
+
+## Usage
+
+Here is an example of the DeleteImageSetConfig:
+
+```yaml
+---
+apiVersion: mirror.openshift.io/v1alpha2
+kind: DeleteImageSetConfiguration
+delete:
+  platform:
+    channels:
+      - name: stable-4.13 
+        minVersion: 4.13.3
+        maxVersion: 4.13.3
+  operators:
+    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.12
+      packages:
+      - name: aws-load-balancer-operator
+  additionalImages: 
+    - name: registry.redhat.io/ubi8/ubi@sha256:bce7e9f69fb7d4533447232478fd825811c760288f87a35699f9c8f030f2c1a6
+    - name: registry.redhat.io/ubi8/ubi-minimal@sha256:8bedbe742f140108897fb3532068e8316900d9814f399d676ac78b46e740e34e
+```
+
+Its immediatley obvious that "kind" and "mirror" fields have changed compared to the ImageSetConfig, this is to ensure
+that all the relevant filtering and validation are done correctly, without having to update the internal code drastically.
+
+The "delete" entry is the main entry, it contains the "platform", "operators" and "additionalImages" entries, these are used to filter the images
+to intentionaly delete the manifests and blobs (in cache only).
+
+### Command line examples
+
+```bash
+
+# NB
+# in both stages --source is mandatory
+# use the --generate flag for stage 1, the flags --config and --delete-id are mandatory
+# the flags --delete-yaml-file and --destination for stage 2 are mandatory --force-cache-delete is optional (default false)
+
+# stage 1
+# -------
+
+# create delete yaml file for delete phase using --generate
+oc-mirror delete --config delete-image-set-config.yaml --source file://<previously-mirrored-work-folder> --v2 --generate --delete-id v4.11
+
+
+# stage 2
+# -------
+
+# once the generate stage has executed succesfully use the created delete yaml to delete the images from the remote registry
+# default setting for the delete-yaml-file is <previously-mirrored-work-folder>/delete/delete-images.yaml
+# delete remote registry only 
+oc-mirror delete --source file://<previously-mirrored-work-folder> --destination docker://<remote-registry> --v2 --delete-yaml-file <previously-mirrored-work-folder>/delete/delete-images-v4.11.yaml 
```

Comment:
interesting choice of using another flag `delete-yaml-file` compared to the generate phase where the flag is `delete-id`. It does force the user to check that file and its contents :thinking:

Reply:
It does feel like --source and --delete-yaml-file are redundant here though. WDYT?